File: idea/testData/hierarchy/calls/callees/javaConstructor/main0.java
class KA {
    KA() {
    }

    public final String name = "A";
    public final String foo(String s) {
        return "A " + s;
    }
}

class KClientBase {

}

class KClient extends KClientBase {
    public <caret>KClient() {
        super();
        new KA().foo(new KA().name);
        new JA().foo(new JA().getName());
    }

    {
        new KA().foo(new KA().name);
        new JA().foo(new JA().getName());
    }

    public final void bar() {
        new KA().foo(new KA().name);
        new JA().foo(new JA().getName());
    }
}

File: idea/testData/hierarchy/overrides/kotlinFunctionInTrait/main0.kt
interface T {
    fun <caret>foo()
}

open class X: T {
    override fun foo() {

    }
}

open class Y: B() {
    override fun foo() {

    }
}

open class Z: X() {
    override fun foo() {

    }
}

class SS {

}